smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/WaiterGenerator.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.kotlin.codegen.rendering.waiters
+
+import software.amazon.smithy.kotlin.codegen.core.KotlinWriter
+import software.amazon.smithy.kotlin.codegen.core.RuntimeTypes
+import software.amazon.smithy.kotlin.codegen.core.withBlock
+import java.text.DecimalFormat
+import java.text.DecimalFormatSymbols
+
+/**
+ * Renders the top-level retry strategy for a waiter.
+ */
+private fun KotlinWriter.renderRetryStrategy(wi: WaiterInfo) {
+    setOf(
+        RuntimeTypes.Core.Retries.Impl.ExponentialBackoffWithJitterOptions,
+        RuntimeTypes.Core.Retries.Impl.ExponentialBackoffWithJitter,
+        RuntimeTypes.Core.Retries.Impl.StandardRetryStrategyOptions,
+        RuntimeTypes.Core.Retries.Impl.StandardRetryStrategy,
+        RuntimeTypes.Core.Retries.Impl.NoOpTokenBucket,
+    ).forEach(::addImport)
+
+    write("")
+    withBlock("private val #L = run {", "}", wi.retryStrategyName) {
+        withBlock("val delayOptions = ExponentialBackoffWithJitterOptions(", ")") {
```

Comment:
Possibly. Right now, `StandardRetryStrategy` uses both a token bucket and a delay provider as part of its implementation. It made sense to me to make those pluggable types so that alternate implementations could be provided (e.g., `NoOpTokenBucket` vs `StandardRetryTokenBucket`). And of course everything has lots of numerical knobs that can be turned like max retries, jitter coefficient, etc. As it is, I can't see a way to reduce the types without collapsing the modularity or configurability of the design.